app/src/main/java/com/door43/translationstudio/newui/home/ImportDialog.java

```diff
@@ -175,16 +187,56 @@ private void showP2PDialog() {
 
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if(requestCode == IMPORT_PROJECT_FROM_SD_REQUEST) {
-            if((resultCode == Activity.RESULT_OK) && (data != null)) {
-                if(isDocumentFile) {
+        if (requestCode == IMPORT_PROJECT_FROM_SD_REQUEST) {
+            if ((resultCode == Activity.RESULT_OK) && (data != null)) {
+                if (isDocumentFile) {
                     Uri uri = data.getData();
                     importUri(uri);
                 } else {
                     File file = new File(data.getData().getPath());
                     importFile(file);
                 }
             }
+        } else if (requestCode == IMPORT_USFM_PROJECT_FROM_SD_REQUEST) {
+            if ((resultCode == Activity.RESULT_OK) && (data != null)) {
+                if (isDocumentFile) {
+                    Uri uri = data.getData();
+                    doUsfmImportUri(uri);
+                } else {
+                    String path = data.getData().getPath();
+                    doUsfmImportFile(path);
+                }
+            }
+        }
+    }
+
+    /**
+     * import USFM uri with fallback to standard import if tstudio uri
+     * @param uri
+     */
+    private void doUsfmImportUri(Uri uri) {
+        String path = uri.toString();
+        String ext = FilenameUtils.getExtension(path).toLowerCase();
+        boolean tstudio = ext.equalsIgnoreCase(Translator.ARCHIVE_EXTENSION);
```

Comment:
I think it would be better to not fallback to importing the archive. If they chose to import usfm we should only allow usfm. Simpler is better.

We should confirm the file extension is `.usfm`.